doc/badge-urls.md

```diff
@@ -0,0 +1,27 @@
+# Badge URL Conventions
+
+- The format of new badges should be of the form `/SERVICE/NOUN/PARAMETERS?QUERYSTRING` e.g:
+  `/github/issues/:user/:repo`. The service is github, the
+  badge is for issues, and the parameters are `:user/:repo`.
+- Parameters should always be part of the route if they are required to display a badge e.g: `:packageName`.
+- Common optional params like, `:branch` or `:tag` should also be passed as part of the route.
+- Query string parameters should be used when:
+  - The parameter is related to formatting. e.g: `/appveyor/tests/:user/:repo?compact_message`.
+  - The parameter is for an uncommon optional attribute, like an alternate registry URL.
+  - The parameter triggers application of alternative logic, like version semantics. e.g: `/github/v/tag/:user/:repo?sort=semver`.
+  - Services which require a url/hostname parameter always should use a query string parameter to accept that value. e.g: `/discourse/topics?server=https://meta.discourse.org`.
+
+It is convention to use the following standard routes and abbreviations across services:
+
+- Coverage: `/coverage`
+- Downloads or Installs:
+  - Total: `/dt`
+  - Per month: `/dm`
+  - Per week: `/dw`
+  - Per day: `/dd`
+  - For version: `/dv`
```

Comment:
I believe ATM this is used in the download badges for gems and crates, but couldn't this be a bit limiting for services that support the daily/weekly/monthly/yearly/etc. download counts both for latest/current version and a specific version? For example if a user wanted the monthly download count of v1.0.0 of package foo.

(I can't think of a service off hand that supports download counts for versions but figured I'd ask anyway 😄)

Reply:
Good question. Given we don't yet have an example of a service where we've done this (AFAIK), there is no convention to document yet.

I think in that case, we would use `dd`/`dw`/`dm` but give all of them an optional version param, so

* `service/dd/mypackage` - downloads per day (all versions)
* `service/dd/mypackage/1.0.0` - downloads per day (version 1.0.0)

as opposed to having `/dd` and `/ddv` (or whatever) as 2 seperate endpoints.

Does that clarify your thinking on what we should document at all?

Reply:
> I think in that case, we would use dd/dw/dm but give all of them an optional version param, so
> 
> * service/dd/mypackage - downloads per day (all versions)
> * service/dd/mypackage/1.0.0 - downloads per day (version 1.0.0)

That's exactly what I had in mind as well. I'd vote for that approach vs. recommending `/dv` for the same future-compatible reasons as recommending `/dt` discussed above

Reply:
Yep sounds reasonable. I've raised #4454 to move the routes we've got in line with this.